app/models/device.rb

```diff
@@ -24,10 +22,18 @@ def self.create_records(file)
   end
 
   def self.top_occurrences(timestamp)
+    # This method returns an ARRAY of arrays! array[0][0] => (device_id)
+    #                                         array[0][1] => (popularity)
+
+    # I think im doing something wrong calling the
+    # .first(10) last in the method chain
```

Comment:
I Don't know how to protect me against all the querys that are going to get fired once I iterate on the hash to find the Device instances according to the devise_id